### PR TITLE
Feat/typography component

### PR DESCRIPTION
**Pull request by Arthurmqz**
This component aims to be the design system typography component

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/components/expandablePanel/src/ExpandablePanel.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  styled,
+  disabled,
+  header,
+  children,
+  expanded,
+  startExpanded,
+  ...props
+}: ExpandablePanelProps): ReactNode => {
+  const theme = useTheme();
+  const [isExpanded, setIsExpanded] = useState(startExpanded || false);
```
Comment (joaoprcosta) [inline packages/components/expandablePanel/src/ExpandablePanel.tsx:23]: Why not set the default value of `startExpanded` to false?

Comment (Arthurmqz) [reply packages/components/expandablePanel/src/ExpandablePanel.tsx]: This is no longer part of the component~

File: packages/components/typography/src/typoTypes/body/Body4.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  const getBody4DefaultThemeValues: TypographyStyledProps = {
+    fontSize: theme.typography.body.body4.mobile.fontSize,
+    lineHeight: theme.typography.body.body4.mobile.lineHeight,
+    fontWeight: theme.typography.body.body4.mobile.fontWeight,
+    color: theme.colors.brand.black,
+    fontFamily: theme.fontFamily.text,
+  };
+
+  const getResponsiveStyles = (): TypographyStyledProps => {
```
Comment (joaoprcosta) [inline packages/components/typography/src/typoTypes/body/Body4.tsx:25]: This method could be a hook. The only difference between each component is the element (e.g. body4) and therefore could be passed as an argument.

Comment (Arthurmqz) [reply packages/components/typography/src/typoTypes/body/Body4.tsx]: I have created the hook and updated the typography components' code

File: packages/react/src/hooks/useResponsiveTypography.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import { BreakpointsType, TypographyStyle, useTheme } from '@douro-ui/react';
```
Comment (joaoprcosta) [inline packages/react/src/hooks/useResponsiveTypography.ts:1]: Inside react package you don't need to point to the package name, just use:

'../'

**Comment by jessywork** [comment]
You can delete the Changelog file

**Review by jessywork — APPROVED** [review]

File: packages/react/src/hooks/__tests__/useHeightDimensions.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    const { result, rerender } = renderHook(
+      ({ isExpanded }: { isExpanded: boolean }) =>
+        useHeightDimension(isExpanded),
+      {
+        initialProps: { isExpanded: false },
+      },
+    );
+
+    act(() => {
```
Comment (joaoprcosta) [inline packages/react/src/hooks/__tests__/useHeightDimensions.spec.tsx:39]: I think `act` is deprecated by the testing-library. You can change to this approach:

```suggestion
    await waitFor(() => ...)
```

File: packages/react/src/hooks/__tests__/useTypography.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+import { useWindowDimensions } from '../useWindowDimensions';
+
+jest.mock('../useTheme');
+jest.mock('../useWindowDimensions');
+jest.mock('../../../../components/utils', () => ({
+  getWindowSize: jest.fn(),
+  getResponsiveTypography: jest.fn(),
+}));
+
+const mockTheme = {
```
Comment (joaoprcosta) [inline packages/react/src/hooks/__tests__/useTypography.spec.tsx:22]: Move this to `tests/__mocks__/` and create a theme.ts to store this.

**Review by joaoprcosta — APPROVED** [review]

**Pull request merged**